cmake/DefaultTargetProperties.cmake

```diff
@@ -0,0 +1,129 @@
+    endif ()
+
+    # Use C++11 ABI by default, but allow disabling.
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
+        CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 5.0)
+        option(GLIBCXX_USE_CXX11_ABI "For gcc, use the new C++11 library ABI" ON)
+        if (GLIBCXX_USE_CXX11_ABI)
+            target_compile_definitions(${target_name} PRIVATE _GLIBCXX_USE_CXX11_ABI=1)
+        else ()
+            target_compile_definitions(${target_name} PRIVATE _GLIBCXX_USE_CXX11_ABI=0)
+        endif ()
+    endif ()
+
+    # TODO(DF): POSITION_INDEPENDENT_CODE required for Python module? See OCIO.
+
+    #-------------------------------------------------------------------
+    # RPATH wrangling.
+
+    # With the 'usual' install path structure in mind, search from the
+    # bin directory (i.e. a binary loading a dynamic library) and then
+    # from the current directory (i.e. dynamic library loading another
+    # dynamic library).
+    if (APPLE)
+        set(rpath "@loader_path/../${CMAKE_INSTALL_LIBDIR};@loader_path")
+    else ()
+        set(rpath "$ORIGIN/../${CMAKE_INSTALL_LIBDIR};$ORIGIN")
+    endif ()
+
+    set_target_properties(
+        ${target_name}
+        PROPERTIES
+        # Allow CMake to set a temporary runtime library search path in
+        # the build phase that potentially hardcodes absolute paths,
+        # useful for TDD.
+        SKIP_BUILD_RPATH FALSE
+        BUILD_WITH_INSTALL_RPATH FALSE
+        # Don't add hardcoded directories to final runtime search path.
+        INSTALL_RPATH_USE_LINK_PATH FALSE
+        # Runtime search path value
+        INSTALL_RPATH "${rpath}"
+        # Enable RPATH on OSX
+        MACOSX_RPATH ON
+    )
+
+    #-------------------------------------------------------------------
+    # C++ #define to toggle static vs dynamic lib
+
+    # Used by header generated by CMake's `generate_export_header`
+    if (NOT BUILD_SHARED_LIBS)
+        target_compile_definitions(${target_name} PRIVATE ${PROJECT_NAME}_STATIC_DEFINE=1)
+    endif ()
+
+
+    #-------------------------------------------------------------------
+    # Build tree directory structure.
+
+    set_target_properties(
+        ${target_name}
+        PROPERTIES
+        # Make the build area layout look a bit more like the final dist
+        # layout
+        LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}
+        RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}
+    )
+
+
+    #-------------------------------------------------------------------
+    # Library version
+
+    # When building or installing appropriate symlinks are created, if
+    # supported.
+    set_target_properties(
+        ${target_name}
+        PROPERTIES
+        VERSION ${PROJECT_VERSION}
+        SOVERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}
+    )
+
+endfunction()
```

Comment:
No blank line at EOF?

Reply:
Added blank lines and updated `.editorconfig` in f81c11d